services/storage/src/simcore_service_storage/datcore.py

```diff
@@ -83,6 +96,71 @@ def list_files_recursively(self, dataset_filter: str=""):
 
         return files
 
+    def list_files_raw_dataset(self, dataset_id: str)->List[FileMetaDataEx]:
+        files = [] # raw packages
+        _files = [] # fmds
+        data = {} # map to keep track of parents-child
+
+        cursor = ''
+        page_size = 1000
+        api = self.client._api.datasets
+
+        dataset = self.client.get_dataset(dataset_id)
+        if dataset is not None:
+            while True:
+                resp = api._get(api._uri('/{id}/packages?cursor={cursor}&pageSize={pageSize}&includeSourceFiles={includeSourceFiles}', id=dataset_id,
+                    cursor=cursor, pageSize=page_size, includeSourceFiles=False))
+                for package in resp['packages']:
```

Comment:
is 'packages' always expected in resp (what if e.g. responds with 500?), otherwise use ``get('packages', list())``.